src/coreclr/src/vm/weakreferencenative.cpp

```diff
@@ -427,29 +485,7 @@ FCIMPL3(void, WeakReferenceNative::Create, WeakReferenceObject * pThisUNSAFE, Ob
     _ASSERTE(gc.pThis->GetMethodTable()->CanCastToClass(pWeakReferenceMT));
 
     // Create the handle.
-#ifdef FEATURE_COMINTEROP
-    IWeakReference* pRawComWeakReference = nullptr;
-    if (gc.pTarget != NULL)
-    {
-        SyncBlock* pSyncBlock = gc.pTarget->PassiveGetSyncBlock();
-        if (pSyncBlock != nullptr && pSyncBlock->GetInteropInfoNoCreate() != nullptr)
-        {
-            pRawComWeakReference = GetComWeakReference(&gc.pTarget);
-        }
-    }
-
-    if (pRawComWeakReference != nullptr)
-    {
-        SafeComHolder<IWeakReference> pComWeakReferenceHolder(pRawComWeakReference);
-        gc.pThis->m_Handle = SetNativeComWeakReferenceHandle(GetAppDomain()->CreateNativeComWeakHandle(gc.pTarget, pComWeakReferenceHolder));
-        pComWeakReferenceHolder.SuppressRelease();
-    }
-    else
-#endif // FEATURE_COMINTEROP
-    {
-        gc.pThis->m_Handle = GetAppDomain()->CreateTypedHandle(gc.pTarget,
-            trackResurrection ? HNDTYPE_WEAK_LONG : HNDTYPE_WEAK_SHORT);
-    }
+    CreateAndSetHandle(&gc.pThis, &gc.pTarget, trackResurrection);
```

Comment:
My usual concern - does this refactoring affect performance for the non-COM case?

Reply:
Boooo - local run was about +2% (creating 10 million weak references was ~2.100s to ~2.145s):
Will put it back.